pkg/autodetect/main.go

```diff
@@ -195,7 +195,10 @@ func (b *Background) detectKafka(ctx context.Context, apiList *metav1.APIGroupLi
 	}
 }
 
-func (b *Background) detectClusterRoles(ctx context.Context) {
+func (b *Background) detectClusterRoles(ctx context.Context, apiList *metav1.APIGroupList) {
+	if !isOpenShift(apiList) {
```

Comment:
At this stage, I think you can do something like this instead:

```go
if viper.GetString("platform") != v1.FlagPlatformOpenShift {
 ...
}
```

This way, you don't need the `apiList`.

Reply:
Thanks! That's looks better, I will update later on today.